src/imitation/algorithms/preference_comparisons.py

```diff
@@ -110,6 +110,7 @@ def __init__(
         self,
         algorithm: base_class.BaseAlgorithm,
         reward_fn: Union[rewards_common.RewardFn, reward_nets.RewardNet],
+        random_frac: float = 0.0,
```

Comment:
This seems like a useful feature, but does it belong in the algorithm? One alternative way of implementing this is to have a wrapper for `BaseAlgorithm` that randomizes between a random agent and the agent being trained. WDYT?

Reply:
That sounds like a good idea because it would make it easy to switch policies within an episode like we discussed. I think we could have two parameters: probability of sticking with the current policy vs randomly picking a new one at each step, and probability of picking each policy (expert or random). That covers purely random trajectories as a special case, so we can replace the current implementation with that.

Unfortunately, I don't think this is really going to simplify the code in `AgentTrainer`. In general, we still need to collect additional expert trajectories, and then collect trajectories with some randomness. Of course we could just take the expert trajectories we already collected during training and fill those up with partially random trajectories, but then the user can't put an upper bound on the proportion of partially random trajectories (we might have collected very few trajectories during training). If you see a way around collecting additional trajectories twice, LMK

Reply:
>  I think we could have two parameters: probability of sticking with the current policy vs randomly picking a new one at each step, and probability of picking each policy (expert or random). That covers purely random trajectories as a special case, so we can replace the current implementation with that.

Seems easy to implement and strictly more general, so I'm in favour.

> Unfortunately, I don't think this is really going to simplify the code in AgentTrainer

You're right. I was thinking for some reason that we were doing the sampling step twice: once from the RL policy, once random. But of course the RL policy is "sampled" as a side-effect of the RL training process (so is effectively free). I think will need to keep current implementation then. I might rename it from `random_frac` to something like `extra_frac` or `exploration_frac` if the trajectories collected by this are not totally random.

Reply:
Implemented the more general version now and renamed to `exploration_frac`